### PR TITLE
Tag to node routes

### PR DESCRIPTION
**Pull request by tobifalowo**
Change the server-side routes to later enable tagging nodes

### CODE REVIEW

**Review by 10hoe — APPROVED** [review]
LGTM

File: src/server/route/path.ts [hunk 1]
```diff
@@ -35,8 +35,20 @@ function path(url: string): IPathRoute {
         '/tag': {
             methods: ['POST', 'GET', 'PUT', 'DELETE'],
         },
-        '/tag/proj/:proj': {
-            methods: ['GET'],
```
Comment (jgke) [inline src/server/route/path.ts:39]: The API should be:
- Get tags for project: GET /tag/:proj
— this should return both tags and the tag-node relations
- Add tag to node: POST /tag/:proj/:node
- Delete tag from node: DELETE /tag/:proj/:node/:tag
Don’t include separate endpoints for creating a tag and adding one, instead do it based on the request — no tagId -> create new label, yes tagId -> use that.

File: src/server/route/routes/tag.ts [hunk 2]
```diff
@@ -108,4 +118,84 @@ router
         res.status(200).json(q);
     });
 
+router.route('/tag/node/tagname').post(async (req: Request, res: Response) => {
+    const projId = req.body.projId;
+    const nodeId = req.body.nodeId;
+    const tagName = req.body.tagName;
+    const tagColor = req.body.tagColor;
```
Comment (jgke) [inline src/server/route/routes/tag.ts:125]: check permissions

File: src/server/route/routes/tag.ts [hunk 3]
```diff
@@ -108,4 +118,84 @@ router
...
+    return res.status(401).json({ error: 'failed to add tagId to node' });
+});
+
+router
+    .route('/tag/node/tagid/remove')
+    .post(async (req: Request, res: Response) => {
+        const projId: number = req.body.projId;
+        const nodeId: number = req.body.nodeId;
+        const tagId: number = req.body.tagId;
+
```
Comment (jgke) [inline src/server/route/routes/tag.ts:177]: Check permissions

File: src/server/route/path.ts (hunk 1 above)
Comment (jgke) [reply src/server/route/path.ts:39]: Also, add api doc annotations.

**Review by jgke — CHANGES_REQUESTED** [review]

**Pull request merged**